pkgs/by-name/co/comfyui/fetch-model.nix

```diff
@@ -0,0 +1,61 @@
+# This is a fetcher made specifically for fetching models to be used by
+# ComfyUI.
+{ lib, pkgs }:
+{ url
+, format
+, bearer ? null
+, bearerFile ? null
+,  ...
+}@args: let
+  name = (lib.replaceStrings
+    ["?" "&" ":" "/"]
+    ["__questionmark__" "__ampersand__" "__colon__" "__slash__"]
+    url
+    # Strictly include the format because some models have multiple components
+    # (such as a pth and yaml file), and the models will often have the same
+    # name, sans extension.
+  ) + "_" + format;
+in {
+  inherit name format;
+  # I think builtins.fetchurl _can_ show progress but needs --verbose to do so.
+  # Need to confirm if this is any better than alternatives (like
+  # nixpkgs.fetchurl).
+  # path = (builtins.fetchurl ({
+  path = (pkgs.fetchurl (
+    (lib.filterAttrs
+      # The @args syntax gathers all arguments, not just the extra ones.
+      # fetchurl doesn't like extra arguments it doesn't know about, so we need
+      # to remove them here.
+      (key: _:
+        (lib.all (x: x != key)
+          ["bearer" "bearerFile" "format"]
+        )
+      )
+      args
+    ) // (lib.optionalAttrs (bearer != null) {
+      # The closest thing to documentation for curlOptsList that I've found:
+      # https://github.com/NixOS/nixpkgs/issues/41820#issuecomment-396120262
+      curlOptsList = [
+        "--header" "Authorization: Bearer ${bearer}"
+        "--location"
+      ];
+    })
+    // (lib.optionalAttrs (bearerFile != null) {
+      # The closest thing to documentation for curlOptsList that I've found:
+      # https://github.com/NixOS/nixpkgs/issues/41820#issuecomment-396120262
+      # This seemed to work at some point but curl can no longer read the file
+      # (but no reason is given).  This is when given world readable
+      # permissions too.
+      curlOptsList = [
+        "--header" "@${bearerFile}"
+        "--location"
```

Comment:
I didn't find any usages of `bearerFile`. Could you elaborate on its purpose too? Also note that if this is some kind of a secret you probably don't want it to go through the nix store

Reply:
Indeed, but some sites require an auth token.  This would be a path to the file, so it wouldn't be in your Nix store.  I should document this better though.  I'm also open to better approaches.